keps/sig-node/0753-sidecarcontainers.md

```diff
@@ -993,264 +438,75 @@ any other container crashes.
 
-We also think that there is some risk of users using `type: sidecar` to any
-sidecar container that doesn't need any startup/shutdown special handling. But
-this is something we can gather feedback during alpha stage and properly
-document. We think that will be enough to alleviate that concern (if anyone
-actually has it) while at the same time achieve the goals this KEP tries to
-achieve.
+## Design Details
 
-All opinions are very welcome on this. Please comment.
+The proposal can broken down into four key pieces of implementation that all relatively separate from one another:
 
-#### Why this design seems extensible?
+* Shutdown triggering for sidecars when RestartPolicy!=Always
+* Sidecars are terminated after normal containers
+* Sidecars start before normal containers
 
-This is a difficult section to add, as different people probably have different
-concerns. So, I'd list the ones that we can think of to open this discussion, and
-encourage everyone to review this critically and see if there is something
-missing to take into account.
+### API Changes:
+As this is a change to the Container spec we will be using feature gating, you will be required to explicitly enable this feature on the api server as recommended [here](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#adding-unstable-features-to-stable-versions).
 
-Let's try to look into the extensibility of this under different scenarios, to
-have a better understanding, but is of course impossible to give guarantees.
+New field `Type` will be added to the lifecycle struct:
 
-##### What if we add pre-defined phases for container startup/shutdown?
+```go
+type Lifecycle struct {
+  // Type
+  // One of Standard, Sidecar.
+  // Defaults to Standard
+  // +optional
+  Type LifecycleType `json:"type,omitempty" protobuf:"bytes,3,opt,name=type,casttype=LifecycleType"`
+}
+```
+New type `LifecycleType` will be added with two constants:
+```go
+// LifecycleType describes the lifecycle behaviour of the container
+type LifecycleType string
 
-This is an invented example. Let's suppose we add a `type: OrderedPhase` in the
-future, that also includes another `Order` field that is an int, starting from
-0.
+const (
+  // LifecycleTypeStandard is the default container lifecycle behaviour
+  LifecycleTypeStandard LifecycleType = "Standard"
```

Comment:
this should be removed, it's either sidecar or nil

Reply:
I think it is ok to keep it. I saw another KEP doing the same and thought it may be clearer. As long as we don't set this field by default (this is what will cause problems), seems fine to treat nil and standard as the same.

I can't find which KEP it was, though... Will update here if I find it.

Anyways, if we don't set Standard by default, do you think it is ok to keep it? Or do you still prefer to remove it?

Reply:
Advice from the API review was to remove it entirely, see https://github.com/kubernetes/kubernetes/pull/79649#discussion_r380919200

Reply:
Ohh, when I read that some weeks ago I didn't get that part. I'll remove it. Thanks!

Reply:
I think this should be fixed now, thanks! :)